Graphviz upstream recently reverted a patch for automatic adoption of
compile/link flags for Perl extensions due to issues on systems where
perl is not available.

This patch brings it back.

https://gitlab.com/graphviz/graphviz/-/commit/b28dd6f8ab3feb00f5f9bff9c8980ac83fac708c
https://gitlab.com/graphviz/graphviz/-/commit/9257d9a19a947435b0f99b9bfd26dfd7fbae413c

--- graphviz-14.1.1/tclpkg/gv/Makefile.am
+++ graphviz-14.1.1/tclpkg/gv/Makefile.am
@@ -99,12 +99,16 @@ gv_lua.cpp: gv.i
 	$(SWIG) -c++ -lua -o $@ $(srcdir)/gv.i
 
 pkgperldir = $(pkglibdir)/perl
+PERL ?= perl
+PERL_CC = `$(PERL) -MExtUtils::Embed -e ccopts`
+PERL_LD = `$(PERL) -MExtUtils::Embed -e ldopts`
+VENDOR_ARCH = `$(PERL) -MConfig -e 'print $$Config{installvendorarch};'`
 PERL_data = gv.pm
 nodist_libgv_perl_la_SOURCES = gv_perl.cpp $(PERL_data)
 libgv_perl_la_SOURCES = $(BASESOURCES) gv_dummy_init.c
 libgv_perl_la_LIBADD = $(BASELIBS) $(PERL_LIBS)
-libgv_perl_la_LDFLAGS = -module -avoid-version
-libgv_perl_la_CPPFLAGS = $(BASECPPFLAGS) $(PERL_INCLUDES)
+libgv_perl_la_LDFLAGS = -module -avoid-version $(PERL_LD)
+libgv_perl_la_CPPFLAGS = $(BASECPPFLAGS) $(PERL_INCLUDES) $(PERL_CC)
 libgv_perl_la_CXXFLAGS = $(AM_CXXFLAGS) -Wno-unused-parameter
 $(PERL_data): gv_perl.cpp
 gv_perl.cpp: gv.i
